src/components/CatalogOverlay/CatalogOverlayComponent.tsx

```diff
@@ -138,18 +135,22 @@ export class CatalogOverlayComponent extends React.Component<WidgetProps> {
     constructor(props: WidgetProps) {
         super(props);
         this.widgetId = "catalog-overlay-0";
-        if (this.widgetStore) {
-            this.catalogFileId = this.widgetStore.catalogInfo.fileId; 
+        const catalogFileId = AppStore.Instance.catalogProfiles.get(this.props.id);
+
+        if (catalogFileId) {
+            AppStore.Instance.catalogProfiles.set(this.props.id, catalogFileId);
```

Comment:
haven't you just retrieved `catalogFileId` from the same place one line before? Why is this `set` needed?
Why not just check if the map `has` `this.props.id`, otherwise setting it to `1`? that would also allow you to remove the `catalogFileId` field, because it's a little confusing to have `this.catalogFileId` and `const catalogFileId`.